### PR TITLE
Add log hooks to catch warnings from DKGProcessor in integration tests

### PR DESCRIPTION
**Pull request by arrivets**
This PR addresses issue [5607](https://github.com/dapperlabs/flow-go/issues/5607) with log hooks

### CODE REVIEW

File: integration/dkg/dkg_whiteboard_test.go [hunk 1]
```diff
@@ -121,6 +121,13 @@ func createNode(
 	)
 	require.NoError(t, err)
 
+	var hook zerolog.HookFunc = func(e *zerolog.Event, level zerolog.Level, message string) {
```
Comment (jordanschalm) [inline integration/dkg/dkg_whiteboard_test.go:124]: I'm OK with this approach, but can we:
* confirm that this is triggered when expected
* add a comment to where these logs are emitted to note that tests depend on that behaviour

Comment (arrivets) [reply integration/dkg/dkg_whiteboard_test.go]: Ok, this test confirms that the hook his triggered when expected:
https://github.com/onflow/flow-go/blob/f2d69135e1969d83d2749efdfefd1df4e9b7add2/module/dkg/broker_test.go#L337-L363

**Review by jordanschalm — APPROVED** [review]

File: integration/dkg/dkg_emulator_suite.go [hunk 2]
```diff
@@ -398,6 +398,15 @@ func (s *DKGSuite) initEngines(node *node, ids flow.IdentityList) {
 	)
 	require.NoError(s.T(), err)
 
+	// We add a hook to the logger such that the test fails if the broker writes
```
Comment (tarakby) [inline integration/dkg/dkg_emulator_suite.go:401]: Nice, just making sure this hook is active only when testing for happy path cases. For unhappy paths, these warnings could happen and it shouldn't make the test fail.

File: integration/dkg/dkg_whiteboard_test.go [hunk 3]
```diff
@@ -121,6 +121,15 @@ func createNode(
 	)
 	require.NoError(t, err)
 
+	// We add a hook to the logger such that the test fails if the broker writes
+	// a Warn log, which happens when it flags or disqualifies a node
```
Comment (tarakby) [inline integration/dkg/dkg_whiteboard_test.go:125]: Same here, just making sure this is only active for happy paths.

File: module/dkg/broker.go [hunk 4]
```diff
@@ -39,6 +39,7 @@ type Broker struct {
 	broadcastMsgCh    chan messages.DKGMessage // channel to forward incoming broadcast messages to consumers
 	messageOffset     uint                     // offset for next broadcast messages to fetch
 	shutdownCh        chan struct{}            // channel to stop the broker from listening
+	broadcasts        uint                     // broadcasts counts the number of succesful broadcasts
```
Comment (tarakby) [inline module/dkg/broker.go:42]:
```suggestion
	broadcasts        uint                     // broadcasts counts the number of successful broadcasts
```

File: module/dkg/broker.go [hunk 5]
```diff
@@ -95,6 +96,11 @@ func (b *Broker) PrivateSend(dest int, data []byte) {
 
 // Broadcast signs and broadcasts a message to all participants.
 func (b *Broker) Broadcast(data []byte) {
+	if b.broadcasts > 0 {
```
Comment (tarakby) [inline module/dkg/broker.go:99]: Do we need a lock here or is the function always called sequentially?

**Review by tarakby — APPROVED** [review]
Nice, thanks for updating the tests 👌

**Pull request merged**